bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/editor/properties/LspEditorPropertiesPage.java

```diff
@@ -48,11 +48,12 @@ public LspEditorPropertiesPage() {
 
 	@Override
 	protected Control createContents(Composite parent) {
+		setPreferenceStore(store());
 		Composite composite = new Composite(parent, SWT.NONE);
 		composite.setLayout(GridLayoutFactory.fillDefaults().numColumns(3).create());
 		composite.setLayoutData(GridDataFactory.fillDefaults().grab(true, false).create());
 		area = new ClangdConfigurationArea(composite, configuration.metadata());
-		area.init(store(), this);
+		area.init(getPreferenceStore(), this);
```

Comment:
When I apply these change, the properties page is empty:
![image](https://github.com/eclipse-cdt/cdt-lsp/assets/123444711/df94c2b2-50ba-4f22-9cf6-f14a9db3e490)

Reply:
I tested with a CMake project. And there are no settings in the .settings folder.

Reply:
Perhaps your workspace settings are already affected by previuos changes. Could you please try to use "Restore Defaults" button on "Properties" dialog?

Reply:
I already created a new workspace folder. And added a new Cmake project. Restore defualts does nothing. You just replaced the `store()` call by `getPreferenceStore()` in `LspEditorProperttiesPage` and changed nothing else?

Reply:
nope, I did also `setPreferenceStore(store());` in the beginning of the method

Reply:
> nope, I did also setPreferenceStore(store()); in the beginning of the method

Sorry my fault!
But the project properties are still overwritten by the workspace preferences. And the `Pretty-print JSON output` can still not be enabled. Neither in WS nor project.

Reply:
Okay. one step forward: I can confirm that the pretty option works, but workspace preferences still overwriting project properties

Reply:
> Checked it with disabling --clang-tidy on project level (while having it enabled for workspace) and opening a .c file from this project initialized clangd as expected.

Can confirm that. But the other way does not work: disable on workspace disables on project level as well

Reply:
Looking further, it seems I found what could be wrong, working to fix this.

Reply:
Fine. Good teamwork!

Reply:
No luck with a kind of "short fix" here, the task we have has no "ready to reuse" solution in the Platform, only some fragments of it.
So, I reimplemented it from scratch and tested for pain points that we already now. Works good for me so far.
![image](https://github.com/eclipse-cdt/cdt-lsp/assets/15242312/1ea62140-cdf5-4418-9aba-96d4e4f37d3d)